ext/datadog_cov/datadog_cov.c

```diff
@@ -5,10 +5,35 @@
 #define DD_COV_TARGET_FILES 1
 #define DD_COV_TARGET_LINES 2
 
+static int is_prefix(VALUE prefix, const char *str)
+{
+  if (prefix == Qnil)
+  {
+    return 0;
+  }
+
+  const char *c_prefix = RSTRING_PTR(prefix);
+  if (c_prefix == NULL)
+  {
+    return 0;
+  }
+
+  long prefix_len = RSTRING_LEN(prefix);
+  if (strncmp(c_prefix, str, prefix_len) == 0)
+  {
+    return 1;
+  }
+  else
+  {
+    return 0;
+  }
+}
+
 // Data structure
 struct dd_cov_data
 {
   VALUE root;
+  VALUE ignored_path;
```

Comment:
So I guess the same suggestion from #171 would apply here -- keeping the prefix as a `char *` and caching its length would save us some per-line work.

Reply:
I agree! If it's not blocking I would merge this one as is, and then optimized it together with everything else in #171

Reply:
I agree! If it's not blocking I would merge this one as is, and then optimized it together with everything else in #171

Reply:
Yeah, I don't think it's a blocker. I guess it's a slight performance regression until it's paired with #171, but I'm guessing the plan will be to release 'em together :)

Reply:
I will have to release it earlier than the performance improvements, but there is an important customer value in this: we have a customer that evaluates ITR and current implementation that does not ignore bundled gems installed in project folder induces up to 400% code coverage overhead.

I measured ineffectiveness of this solution using test-environment: for the worst case it increases overhead by 7% (113 vs 106) if bundled gems are installed in project folder. When gems are not installed in project folder, this change does not induce any overhead.

Reply:
Sounds reasonable, I'll defer to your wise judgment here :)